### ISSUE TITLE
YOLOv11: Poor results and blurry previews when training on 16-bit single-channel images

### ISSUE THREAD

**Issue by AppleBella123** [ultralytics/ultralytics#21645]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

My custom dataset consists of **16-bit, single-channel** images.

However, I want to investigate whether it is feasible to keep the data in 16-bit, single-channel format.

I found that when I set **channels=1 in the YAML file**:

The preview images generated during training look very different from the original images, and they appear blurry.

If I convert the images to **8-bit, 3-channel** format and set **channels=3** in the YAML file, training works normally.

Questions:

- Does YOLOv11 officially support training on 16-bit, single-channel images without converting them to 8-bit RGB?
- 
- If 16-bit, single-channel input is supported, what is the recommended preprocessing pipeline to ensure correct training and visualization?
I am uploading three images for reference:

- One original image
- ![Image](https://github.com/user-attachments/assets/cf6a730c-0850-48df-974c-399f0d4cfdce)
- One 16-bit single-channel training preview image
- ![Image](https://github.com/user-attachments/assets/2ab0a60d-18a8-4a54-ba80-832c6ce6ba2b)
- One 8-bit 3-channel training preview image
- ![Image](https://github.com/user-attachments/assets/3461ced8-445b-4161-b55b-00ae8cb0ccc1)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AppleBella123, thank you for your detailed report and for using Ultralytics 🚀 This is an automated response to help you get moving quickly — an Ultralytics engineer will also review and assist you soon.

We recommend a visit to the Docs for guidance where you can find many Python and CLI usage examples and answers to common questions:
- Docs: https://docs.ultralytics.com/
- Python examples: https://docs.ultralytics.com/usage/python/
- CLI examples: https://docs.ultralytics.com/usage/cli/

If this is a 🐛 Bug Report, please provide a minimum reproducible example (MRE) to help us debug it: https://docs.ultralytics.com/help/minimum-reproducible-example/. For issues related to data formats and preprocessing, please include:
- A small sample of images illustrating 16-bit single-channel vs 8-bit 3-channel behavior
- The exact YAML (including channels, imgsz, and augmentations)
- Your training command and environment details
- A short code snippet of your preprocessing/loader transforms
- Console logs and any warnings, plus screenshots of the training previews

If this is a custom training ❓ Question, please share as much context as possible (dataset samples and training logs) and verify you’re following our Tips for Best Training Results: https://docs.ultralytics.com/guides/model-training-tips/

Join the Ultralytics community where it suits you best:
- Real-time chat on Discord 🎧: https://discord.com/invite/ultralytics
- In-depth discussions on Discourse: https://community.ultralytics.com/
- Community threads on our Subreddit: https://www.reddit.com/r/Ultralytics/

## Upgrade

Please upgrade to the latest ultralytics package including all requirements in a Python>=3.8 environment with PyTorch>=1.8 to verify the issue isn’t already resolved in the latest version:
```bash
pip install -U ultralytics
```
- Requirements: https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):
- Notebooks with free GPU:
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>
  <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
  <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all Ultralytics CI tests are currently passing. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit:
- Modes: https://docs.ultralytics.com/modes/
- Tasks: https://docs.ultralytics.com/tasks/

To help our team specifically investigate your 16-bit single-channel scenario and the blurry previews in YOLO11, please add:
1. Your data YAML with channels set and image sizes
2. A minimal sample (2–3 images + labels) that reproduces the preview issue
3. Exact training command and version info (ultralytics, PyTorch, CUDA)
4. Any custom dataloader/transform steps used prior to training
5. Screenshots of the training previews you observed (you already attached these — thank you! 🙌)

We’ll review and follow up shortly.

**Comment by Y-T-G** [comment]
Training with 16-bit images isn't supported. It gets converted to 8-bit while reading.

**Comment by AppleBella123** [comment]
> Training with 16-bit images isn't supported. It gets converted to 8-bit while reading.

which file can be modified if I want to read 16-bit images？ Thx.

**Comment by Y-T-G** [comment]
8-bit expectation is hardcoded in several places. You will need to change all the scaling by `255` to `65535`:

https://github.com/search?q=repo%3Aultralytics%2Fultralytics+255+language%3APython&type=code&l=Python&p=2
